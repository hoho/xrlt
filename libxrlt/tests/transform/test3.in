id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:"hello
id:1, type:2, last:1, error: 0, data: world"
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:1, error: 0, data:some
id:2, type:2, last:1, error: 0, data:<param><value>awe</value></param>
id:0, type:0, last:0, error: 0, data:
id:4, type:2, last:1, error: 0, data:{"this": "is", "pretty": "amazing"}
id:0, type:0, last:0, error: 0, data:
