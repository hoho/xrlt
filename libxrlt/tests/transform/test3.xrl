<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">
    <xrl:response>
        <xrl:variable name="href">
            <xrl:include>
                <xrl:href>/test/href/1</xrl:href>
                <xrl:type>json</xrl:type>
                <xrl:success select="/" />
            </xrl:include>
        </xrl:variable>

        <xrl:variable name="par1">
            <xrl:include>
                <xrl:href>/test/href/2</xrl:href>
                <xrl:type>xml</xrl:type>
                <xrl:success select="/param/value" />
            </xrl:include>
        </xrl:variable>

        <xrl:variable name="par2">
            <xrl:include>
                <xrl:href>/test/href/3</xrl:href>
                <xrl:type>text</xrl:type>
            </xrl:include>
        </xrl:variable>

        <hi>yo</hi>

        <xrl:include>
            <xrl:href select="$href" />
            <xrl:type>json</xrl:type>
            <xrl:with-param name="p" select="concat($par1, $par2)" />
        </xrl:include>

        <bye>yep</bye>
    </xrl:response>
</xrl:requestsheet>
