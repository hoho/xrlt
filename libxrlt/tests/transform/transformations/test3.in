id:0, type:100, last:0, error:0, data:
id:0, type:100, last:0, error:0, data:
id:1, type:400, last:0, error:0, data:200
id:1, type:600, last:1, error:0, data:{"data": [{"name": "test2", "item": [1, 2, 3]}, {"name": "test3", "item": ["a", "b", "c"]}]}
id:0, type:100, last:0, error:0, data:
id:3, type:400, last:0, error:0, data:200
id:3, type:600, last:1, error:0, data:=
id:0, type:100, last:0, error:0, data:
id:4, type:400, last:0, error:0, data:200
id:4, type:600, last:1, error:0, data:|
id:0, type:100, last:0, error:0, data:
id:5, type:400, last:0, error:0, data:200
id:5, type:600, last:1, error:0, data:\
id:2, type:400, last:0, error:0, data:200
id:2, type:600, last:1, error:0, data:--
id:0, type:100, last:0, error:0, data:
