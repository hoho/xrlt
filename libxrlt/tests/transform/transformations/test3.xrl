<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">

    <xrl:transformation name="json-stringify" type="json-stringify" />
    <xrl:transformation name="xslt-stringify" type="xslt-stringify" src="test4.xsl" />

    <xrl:response>
        <xrl:variable name="tmptmp" select="'-'" />

        <xrl:include>
            <xrl:href>
                <xrl:text>/tmp</xrl:text>
            </xrl:href>
            <xrl:type>json</xrl:type>

            <xrl:success>
                <xrl:transform name="json-stringify">
                    <test1>
                        <xrl:transform name="xslt-stringify">
                            <test2>
                                <xrl:for-each select="/data[name = 'test2']/item">
                                    <xrl:value-of select="$tmptmp" />
                                    <xrl:variable name="testi">
                                        <xrl:include>
                                            <xrl:href>/tmpi</xrl:href>
                                            <xrl:type>text</xrl:type>
                                        </xrl:include>
                                    </xrl:variable>

                                    <item>
                                        <xrl:copy-of select="concat($testi, text(), $testi)" />
                                    </item>
                                </xrl:for-each>
                            </test2>
                            <test3>
                                <xrl:for-each select="/data[name = 'test3']/item">
                                    <item2>
                                        <xrl:copy-of select="concat('f', text(), '-')" />
                                    </item2>
                                </xrl:for-each>
                            </test3>
                        </xrl:transform>
                    </test1>
                </xrl:transform>
            </xrl:success>
        </xrl:include>
    </xrl:response>

</xrl:requestsheet>
