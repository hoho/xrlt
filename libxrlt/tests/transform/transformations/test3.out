XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: JSON
sr url: /tmp
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 2
sr method: GET
sr type: TEXT
sr url: /tmpi2
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_SUBREQUEST
sr id: 3
sr method: GET
sr type: TEXT
sr url: /tmpi
sr query: (null)
sr body: (null)
XRLT_STATUS_SUBREQUEST
sr id: 4
sr method: GET
sr type: TEXT
sr url: /tmpi
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_SUBREQUEST
sr id: 5
sr method: GET
sr type: TEXT
sr url: /tmpi
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: {"test1": "<test2><item>=1=</item><item>|2|</item><item>\\3\\</item></test2><test3><item2>--a-</item2><item2>--b-</item2><item2>--c-</item2></test3>\n"}
XRLT_STATUS_DONE
